llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -553,28 +801,36 @@ void processFunction(Function &F, FunctionToAspectsMapTy &FunctionToUsedAspects,
       getAspectsFromType(F.getReturnType(), TypesWithAspects);
   for (const auto &Aspect : RetTyAspects)
     if (!FP64ConvEmu || (Aspect != FP64Aspect) ||
-        !hasDoubleType(F.getReturnType()))
-      FunctionToUsedAspects[&F].insert(Aspect);
+        !hasDoubleType(F.getReturnType())) {
+      if (FunctionIsUnconditional)
+        FunctionToUsedAspects[&F].insert(Aspect);
+      if (FunctionIsConditional)
+        FunctionToConditionallyUsedAspects[&F].insert(Aspect);
```

Comment:
Why don't we have `else` here? Does it mean that a function could be both/neither conditional and unconditional at the same time?